enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,773 @@
+            PodPlacementController ->> PodPlacementController: Intersect Supported Architectures
+            PodPlacementController ->> Pod: PatchNodeAffinity
+            Note over PodPlacementController: inspection errors are ignored<br>to allow the removal of the scheduling<br>gate.
+            deactivate PodPlacementController
+            PodPlacementController ->> Pod: RemoveSchedulingGate
+        end
+    end
+    Note over Scheduler: no more schedulingGates
+    Scheduler ->> Scheduler: move Pod to activeQ <br> and other plugins
+
+```
+#### Deletion of the CR
+1. The cluster admin deletes the CR
+2. All resources related to the operator - the controllers, webhook service and
+   pods automatically get deleted
+3. Any pods stuck in the scheduling gated phase will have to be patched manually
+   to resume scheduling (by removing the scheduling gate)
+4. Recreating the CR again will install all related resources
+
+#### Operator uninstallation
+- Operator uninstallation removes the operator. If the option to uninstall
+  operands is selected, all resources are removed. 
+
+#### Variation
+- *Pods ignored during scheduling*: the `PodPlacementConfig` CR allows a
+  namespace exclusion list. Any pods belonging to the excluded namespaces will
+  be gated at admission time and will be ignored by the `PodPlacementController`
+- *NodeAffinity block already set in the gated pod*: when patching the node
+  affinity, any node selector term having a `matchExpression` element using the
+  key `kubernetes.io/arch` is preserved. For all the other node
+  selector terms, we append the architecture-specific `matchExpression`.
+- *Image inspection failure*: in the node affinity patching workflow, if there
+  are any errors during the image inspection process (authentication issues,
+  network errors, ...), the scheduling gate is removed, and no changes are made
+  to the pod. In this case, the scheduling will only be architecture-aware if
+  the user submitting the pod defines the affinity predicates explicitly.
+
+### API Extensions
+
+#### Mutating Admission Webhook and Scheduling Gate 
+The multiarch-manager-operator reconciles not-yet-scheduled pods using the
+scheduling readiness[^1] and pod mutable scheduling directives[^2] features.
+Since Kubernetes 1.27[^3] promoted those features to beta, no additional changes
+to the Openshift Cluster Operators are required to leverage them in the context
+of automatically setting the architecture-related node affinity of pods. As
+described in the Workflow Description section, the
+`SchedulingGateMutatingAdmissionWebhook` changes a pod at admission time. It
+adds the `multi-arch.openshift.io/arch-aware-placement` scheduling gate to newly
+created pods. The scheduling gate can only be added to pods at creation time.
+The `PodPlacementController` is responsible for deleting the scheduling gate
+from the pod after inspecting the architecture of the pod containers' images and
```

Comment:
we have proposed fail open now - just to keep the behavior equivalent to not having the operator installed at all. If the image inspection fails, the pod is updated with the gate removed  in which case it might be scheduled on a node where the image cannot run and crashloopbackoff - the reason being that the pod can always be manually patched with the correct affinities. I am not sure if there are arguments for blocking the scheduling by keeping the gate and failing in which case manual intervention would be required as well. Such would be the case if a pod absolutely should not land on the inappropriate arch node for some reason.

Reply:
Some are already at https://github.com/openshift/enhancements/pull/1417/files#diff-4fdb9132744e05eb241500caf281668871c3b14b3ebd6eee8722774e07cbd33eR546 . We will make it clearer.

Reply:
Clarified on the last commit.